### PR TITLE
Make logic for students per country and statistics level

### PR DESCRIPTION
**Pull request by DmytryiStriletskyi**
Although changes in header of pull request, it contains changed logic for taking first (only one) token, refactor core code for instance`s data.

### CODE REVIEW

File: openedx/core/djangoapps/edx_global_analytics/tasks.py [hunk 1]
```diff
@@ -45,17 +48,20 @@ def count_data():
     # 30th day <= Current last login date <= Today
     activity_border = (datetime.datetime.now() - datetime.timedelta(days=olga_settings.get("ACTIVITY_PERIOD")))
     active_students_amount = UserProfile.objects.all().exclude(
-        user__last_login=None).filter(user__last_login__gt=activity_border).count()
+        Q(user__last_login=None) | Q(user__is_active=False)).filter(user__last_login__gt=activity_border).count()
+
+    # Aggregated students per country
+    students_per_country = UserProfile.objects.exclude(Q(user__last_login=None) | Q(user__is_active=False)).filter(
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: in upper example you have all(), and here you do not have all(), is it ok?

File: openedx/core/djangoapps/edx_global_analytics/tasks.py [hunk 2]
```diff
@@ -78,14 +84,27 @@ def count_data():
...
+        'latitude': latitude,
+        'level': 'enthusiast',
+        'longitude': longitude,
+        'platform_name': platform_name,
+        'platform_url': platform_url,
+        'secret_token': secret_token,
+        'students_per_country': json.dumps(list(students_per_country))
+        }
+
+    paranoid = {
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: use paranoid as main, and append enthusiast additional fields

File: openedx/core/djangoapps/edx_global_analytics/tasks.py [hunk 3]
```diff
@@ -78,14 +84,27 @@ def count_data():
...
+        'active_students_amount': active_students_amount,
+        'courses_amount': courses_amount,
+        'level': 'paranoid',
+        'secret_token': secret_token
+        }
+
     if statistics_level == 1:
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: better use string constants for level, or show what is 1 and what is 2

File: openedx/core/djangoapps/edx_global_analytics/tasks.py (hunk 1 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: It`s harmless mistake - when I use filter() or exclude() I already have queryset, that is the same with all().
Methods above proxies to queryset. I will remove it, thanks.

File: openedx/core/djangoapps/edx_global_analytics/tasks.py (hunk 2 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: Done.

File: openedx/core/djangoapps/edx_global_analytics/tasks.py (hunk 3 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: I did comments.

File: openedx/core/djangoapps/edx_global_analytics/tasks.py [hunk 4]
```diff
@@ -11,11 +11,14 @@
 
 from django.conf import settings
 from django.contrib.sites.models import Site
-from django.core.exceptions import ObjectDoesNotExist
 from xmodule.modulestore.django import modulestore
+
 from student.models import UserProfile
 from .models import TokenStorage
 
+from django.db.models import Count
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: Please fix order of imports

File: openedx/core/djangoapps/edx_global_analytics/tasks.py [hunk 5]
```diff
@@ -44,18 +47,21 @@ def count_data():
     # Built-in edX`s accounts have None in `last_login` field.
     # 30th day <= Current last login date <= Today
     activity_border = (datetime.datetime.now() - datetime.timedelta(days=olga_settings.get("ACTIVITY_PERIOD")))
-    active_students_amount = UserProfile.objects.all().exclude(
-        user__last_login=None).filter(user__last_login__gt=activity_border).count()
+    active_students_amount = UserProfile.objects.exclude(
+        Q(user__last_login=None) | Q(user__is_active=False)).filter(user__last_login__gt=activity_border).count()
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: please follow open edx style.

File: openedx/core/djangoapps/edx_global_analytics/tasks.py [hunk 6]
```diff
@@ -44,18 +47,21 @@ def count_data():
...
+
+    # Aggregated students per country
+    students_per_country = UserProfile.objects.exclude(Q(user__last_login=None) | Q(user__is_active=False)).filter(
+        user__last_login__gt=activity_border).values('country').annotate(count=Count('country'))
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: same here about style

File: openedx/core/djangoapps/edx_global_analytics/tasks.py [hunk 7]
```diff
@@ -78,14 +84,28 @@ def count_data():
...
             'platform_url': platform_url,
-            'secret_token': secret_token
-            })
+            'statistics_level': 'enthusiast',
+            'students_per_country': json.dumps(list(students_per_country))
+        })
+        requests.post(post_url, enthusiast)
+
+    # Paranoid level
+    elif statistics_level == 2:
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: better move it to the 1st check

File: openedx/core/djangoapps/edx_global_analytics/tasks.py [hunk 8]
```diff
@@ -78,14 +84,28 @@ def count_data():
         platform_name = settings.PLATFORM_NAME
     else:
         platform_name = Site.objects.get_current()
- 
+
+    # Sending instance data
+    paranoid = {
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: i think we can just name it data, and paranoid will be in the settings, and there will be no need to copy

File: openedx/core/djangoapps/edx_global_analytics/tasks.py (hunk 7 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: Done.

File: openedx/core/djangoapps/edx_global_analytics/tasks.py (hunk 8 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: I understood, done.

File: openedx/core/djangoapps/edx_global_analytics/tasks.py (hunk 6 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: Done.

File: openedx/core/djangoapps/edx_global_analytics/tasks.py (hunk 4 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: Done.

File: openedx/core/djangoapps/edx_global_analytics/tasks.py [hunk 9]
```diff
@@ -44,18 +46,24 @@ def count_data():
...
-    active_students_amount = UserProfile.objects.all().exclude(
-        user__last_login=None).filter(user__last_login__gt=activity_border).count()
+    active_students_amount = UserProfile.objects.exclude(
+        Q(user__last_login=None) | Q(user__is_active=False)
+    ).filter(user__last_login__gt=activity_border).count()
+
+    # Aggregated students per country
+    students_per_country = UserProfile.objects.exclude(
+        Q(user__last_login=None) | Q(user__is_active=False)
+    ).filter(user__last_login__gt=activity_border).values('country'
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: you can probably save ```UserProfile.objects.exclude(Q(user__last_login=None) | Q(user__is_active=False))``` to variable and then use it in calculation of active_students_amount and students_per_country. But it is as you decide, you can skip this comment for now.

File: openedx/core/djangoapps/edx_global_analytics/tasks.py [hunk 10]
```diff
@@ -78,14 +86,27 @@ def count_data():
...
-            'courses_amount': courses_amount,
-            'active_students_amount': active_students_amount,
+
+    # Sending instance data
+    data = {
+        'active_students_amount': active_students_amount,
+        'courses_amount': courses_amount,
+        'statistics_level': 'paranoid',
+        'secret_token': secret_token
+        }
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: Maybe wrong indent of closing bracket.

Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: Fixed.

File: openedx/core/djangoapps/edx_global_analytics/tasks.py (hunk 9 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: I want to save my own logic for this stage of development, but surely I`ll make best logic in future if needed, thanks.

Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: 👍

File: openedx/core/djangoapps/edx_global_analytics/tasks.py [hunk 11]
```diff
@@ -78,14 +86,27 @@ def count_data():
...
             'latitude': latitude,
             'longitude': longitude,
             'platform_name': platform_name,
             'platform_url': platform_url,
-            'secret_token': secret_token
-            })
+            'statistics_level': 'enthusiast',
+            'students_per_country': json.dumps(list(students_per_country))
+        })
+        requests.post(post_url, data)
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: PLease move it out of if else

Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: Done.

**Pull request merged**